ANW-1549: display indicator that top container has restricted material

avatar382 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
ANW-1549: display indicator that top container has restricted material

## Related JIRA Ticket or GitHub Issue
https://archivesspace.atlassian.net/jira/software/c/projects/ANW/issues/ANW-1549

## How Has This Been Tested?
manual in browser testing

## Screenshots (if appropriate):
![Screenshot_2022-05-11_15-06-42](https://user-images.githubusercontent.com/130926/167958916-d40bc5b4-c1dc-4d38-8c8d-c0a8fb21573d.jpg)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

mark-cooper commented:
@avatar382 the `staff_access?` method stuff in resources_controller looks out of context. Is it supposed to be here?

avatar382 commented:
> @avatar382 the `staff_access?` method stuff in resources_controller looks out of context. Is it supposed to be here?

I've renamed the method to remove the "?" since it's not returning a boolean and moved it to application_controller.rb with the other controller helper methods.

mark-cooper commented on public/app/controllers/resources_controller.rb:
```diff
@@ -243,6 +243,7 @@ def tree_node_from_root
 
   def inventory
     uri = "/repositories/#{params[:rid]}/resources/#{params[:id]}"
+    check_staff_access(uri)
```
@avatar382 if I delete this line (and remove the `check_staff_access` method altogether) then the functionality is exactly the same because the visibility is determined by the `smart_staff_links` js. So I think this can be chopped out, right?

avatar382 replied:
@mark-cooper Yes -- good catch, I added that while trying to figure out how the the logged in access works.

I've removed all references to the `check_staff_access` method and rebased with the latest master.
